### PR TITLE
Sort segments on timestamp in read only engine

### PR DESCRIPTION
**Pull request by javanna**
Ordinary indices support sorting their segments on the timestamp when they have such a field defined in their mappings (see #75195). This was not initially supported as a directory reader could not be created provding a leaf sorter, which is now possible in Lucene and we can make use of.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/sort_segments_on_timestamp.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      indices.create:
+        index: test
+        body:
+          mappings:
+            properties:
+              "@timestamp":
+                type: date
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
```
Comment (tlrx) [inline x-pack/plugin/searchable-snapshots/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/sort_segments_on_timestamp.yml:14]: Can we rely on the default number of replicas?

**Review by tlrx — APPROVED** [review]
LGTM, Thanks Luca!

File: x-pack/plugin/searchable-snapshots/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/sort_segments_on_timestamp.yml (hunk 1 above)
Comment (javanna) [reply x-pack/plugin/searchable-snapshots/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/sort_segments_on_timestamp.yml:14]: makes sense, shall I also remove setting the number of primary shards?

Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/sort_segments_on_timestamp.yml:14]: No, I think the test can only work with `number_of_shards: 1` and as such it should be skipped for non default shard number. See `features: default_shards`

**Pull request merged**